### PR TITLE
MG-2161 - Generate mocks with mockery for mqtt

### LINKED ISSUE #2161: Generate mocks with mockery for mqtt
### Is your feature request related to a problem? Please describe.
Yes, the feature request is related to the problem of testing mqtt. Currently, our tests are not isolated from the implementation details which can lead to brittle tests.

### Describe the feature you are requesting, as well as the possible use case(s) for it.
We are requesting the feature of mock generation for mqtt. This feature will allow us to write unit tests that are isolated from the implementation details. The use case for this feature is primarily in testing - it will allow us to write more robust and reliable unit tests.

### Indicate the importance of this feature to you.
Must-have

### PR DESCRIPTION
**Pull request by JeffMboya**
# What type of PR is this?

This is a feature because it adds mocks for mqtt using mockery.

## What does this do?

This PR introduces mock generation for the mqtt. This will allow us to write unit tests that are isolated from the implementation details. 

## Which issue(s) does this PR fix/relate to?

- Related Issue #2161 
- Resolves #2161

## Have you included tests for your changes?

Yes

## Did you document any new/modified feature?

No

### CODE REVIEW

File: mqtt/events/streams.go [hunk 1]
```diff
@@ -12,6 +12,7 @@ import (
 
 const streamID = "magistrala.mqtt"
 
+//go:generate mockery --name EventStore --output=../mocks --filename redis.go --quiet --note "Copyright (c) Abstract Machines"
```
Comment (rodneyosodo) [inline mqtt/events/streams.go]: Add it to CI

File: mqtt/handler_test.go [hunk 2]
```diff
@@ -107,17 +107,19 @@ func TestAuthConnect(t *testing.T) {
 	}
 
 	for _, tc := range cases {
+		repoCall := eventStore.On("Connect", mock.Anything, mock.Anything).Return(nil)
```
Comment (rodneyosodo) [inline mqtt/handler_test.go]: Try and reduce the use of `mock.Anything` also this is not a `repocall`

**Review by rodneyosodo — CHANGES_REQUESTED** [review]

File: .github/workflows/api-tests.yml [hunk 3]
```diff
@@ -183,4 +183,4 @@ jobs:
 
       - name: Stop containers
         if: always()
-        run: make run down args="-v"
+        run: make run down args="-v"
```
Comment (rodneyosodo) [inline .github/workflows/api-tests.yml]: Add a new line

File: mqtt/events/streams.go (hunk 1 above)
Comment (rodneyosodo) [inline mqtt/events/streams.go]:
```suggestion
//go:generate mockery --name EventStore --output=../mocks --filename events.go --quiet --note "Copyright (c) Abstract Machines"
```

File: .github/workflows/check-generated-files.yml [hunk 4]
```diff
@@ -164,3 +166,4 @@ jobs:
           check_mock_changes ./bootstrap/mocks/configs.go "Bootstrap Repository ./bootstrap/mocks/configs.go"
           check_mock_changes ./invitations/mocks/service.go "Invitations Service ./invitations/mocks/service.go"
           check_mock_changes ./invitations/mocks/repository.go "Invitations Repository ./invitations/mocks/repository.go"
+          check_mock_changes ./mqtt/mocks/redis.go "MQTT Redis ./mqtt/mocks/redis.go"
```
Comment (rodneyosodo) [inline .github/workflows/check-generated-files.yml]:
```suggestion
          check_mock_changes ./mqtt/mocks/redis.go "MQTT Events Store ./mqtt/mocks/redis.go"
```

File: mqtt/handler_test.go [hunk 5]
```diff
@@ -61,10 +61,11 @@ var (
 		Username: invalidID,
 		Password: []byte(password),
 	}
+	svcCall *mock.Call
```
Comment (rodneyosodo) [inline mqtt/handler_test.go]: Remove this top-level variable definition

File: mqtt/handler_test.go [hunk 6]
```diff
@@ -433,21 +439,27 @@ func TestDisconnect(t *testing.T) {
...
-func newHandler() (session.Handler, *authmocks.AuthClient) {
+func newHandler(t *testing.T) (session.Handler, *authmocks.AuthClient, *mocks.EventStore) {
 	logger, err := mglog.New(&logBuffer, "debug")
 	if err != nil {
 		log.Fatalf("failed to create logger: %s", err)
 	}
 	auth := new(authmocks.AuthClient)
-	eventStore := mocks.NewEventStore()
-	return mqtt.NewHandler(mocks.NewPublisher(), eventStore, logger, auth), auth
+	eventStore := mocks.NewEventStore(t)
```
Comment (rodneyosodo) [inline mqtt/handler_test.go]:
```suggestion
	eventStore := new(mocks.EventStore)
```

File: mqtt/handler_test.go [hunk 7]
```diff
@@ -105,19 +106,24 @@ func TestAuthConnect(t *testing.T) {
 			session: &sessionClient,
 		},
 	}
-
 	for _, tc := range cases {
 		ctx := context.TODO()
+		if tc.session != nil {
+			svcCall = eventStore.On("Connect", mock.Anything, string(tc.session.Password)).Return(tc.err)
+		} else {
```
Comment (rodneyosodo) [inline mqtt/handler_test.go]: If session is not in context it should return `ErrClientNotInitialized`

**Review by rodneyosodo — CHANGES_REQUESTED** [review]

**Review by rodneyosodo — APPROVED** [review]
LGTM

**Review by dborovcanin — APPROVED** [review]

**Pull request merged**